scripts/build/check_init_priorities.py

```diff
@@ -120,60 +121,26 @@ class variable.
     def __init__(self, file_path):
         self.file_path = file_path
         self._elf = ELFFile(open(file_path, "rb"))
-        self._load_symbols()
         self._find_defined_devices()
 
-    def _load_symbols(self):
-        """Initialize the symbols table."""
-        self._symbols = {}
-
-        for section in self._elf.iter_sections():
-            if not isinstance(section, SymbolTableSection):
-                continue
-
-            for num, sym in enumerate(section.iter_symbols()):
-                if sym.name:
-                    self._symbols[num] = sym.name
-
-    def _device_ord_from_rel(self, rel):
-        """Find a device ordinal from a device symbol name."""
-        sym_id = rel["r_info_sym"]
-        sym_name = self._symbols.get(sym_id, None)
-
-        if not sym_name:
-            return None
-
-        if not sym_name.startswith(_DEVICE_ORD_PREFIX):
-            return None
-
-        _, device_ord = sym_name.split(_DEVICE_ORD_PREFIX)
-        return int(device_ord)
-
     def _find_defined_devices(self):
-        """Find the device structures defined in the object file."""
-        self.defined_devices = {}
+        dev_syms = []
+        rel_sects = {}
+        for i, section in enumerate(self._elf.iter_sections()):
```

Comment:
Would it work to use `enumerate(self._elf.iter_sections(), 1)` and drop the `+ 1` in line 142?

Reply:
added, was not aware of this feature of enumerate...

Reply:
actually this was wrong, posted a new revision